From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <spottedleaf@spottedleaf.dev>
Date: Sat, 16 Jan 2021 08:18:51 -0800
Subject: [PATCH] Do not add passengers of entities that were were above save
 limit

Given that the root entity isn't added to the world, this is
pretty unsafe to do.

diff --git a/src/main/java/net/minecraft/server/ChunkRegionLoader.java b/src/main/java/net/minecraft/server/ChunkRegionLoader.java
index 5082ee57c0f4cbba0cca73ae9c9ae4b4c8a53ced..d0bc446815006cf1b527d1e51e051677b05ec19f 100644
--- a/src/main/java/net/minecraft/server/ChunkRegionLoader.java
+++ b/src/main/java/net/minecraft/server/ChunkRegionLoader.java
@@ -659,7 +659,7 @@ public class ChunkRegionLoader {
                 final int saveLimit = world.paperConfig.entityPerChunkSaveLimits.getOrDefault(entityType, -1);
                 if (saveLimit > -1) {
                     if (loadedEntityCounts.getOrDefault(entityType, 0) >= saveLimit) {
-                        return entity;
+                        return null; // Tuinity - do not merge potentially unsafe passenger state into world
                     }
                     loadedEntityCounts.merge(entityType, 1, Integer::sum);
                 }
